src/main/java/com/onelogin/sdk/conn/Client.java

```diff
@@ -2930,8 +2930,27 @@ public Boolean verifyFactorOtp(long userId, String verificationId, String otp, S
 		params.put("otp", otp);
 		params.put("device_id", deviceId);
 		
-		return updateOperation(params, url, versionId);
-    }
+		//Modified updateOperation  
+		String body = JSONUtils.buildJSON(params);
+   		if (versionId == 1) {
+			//fail if using version 1
+			return false;
+		} else {
+			OneloginOAuth2JSONResourceResponse oAuth2Response = (OneloginOAuth2JSONResourceResponse) executeCall(url, OAuth.HttpMethod.PUT, OneloginOAuth2JSONResourceResponse.class, body);
+			//fail if don't get 200 OK response
+			if (oAuth2Response.getResponseCode() != 200) {
+				setResponseError(oAuth2Response, true);
+				return false;
+			} else {
+				//If 200 response check if otp was accepted
+				if (oAuth2Response.getStatus().equals("accepted")) {
```

Comment:
The `params` object that this pulls status from is only set when the response is not 200, 201, or 204 so I modified this code to just pull it out of the content its self.